inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/tasks/SelectionTask.java

```diff
@@ -193,7 +195,7 @@ public void run()
                             recommender, user.getUsername(), result,
                             System.currentTimeMillis() - start, activated));
                 }
-                catch (Throwable e) {
+                catch (ConcurrentException | RecommendationException e) {
```

Comment:
See this [comment](https://github.com/inception-project/inception/issues/1430#issuecomment-538633971). Catching `Throwable` is intended and by design.

Reply:
Dear teacher, what do I need to do next?  Know the reason why a Throwable is used instead of specific exceptions. Close pull request?

Reply:
No, you can improve the pull request as follows. Keep the change in `PDFExtractor.java`, this one is good (see [comment](https://github.com/inception-project/inception/pull/1431#discussion_r333212781)). 
For the other files, follow this [comment](https://github.com/inception-project/inception/pull/1431#discussion_r333211582), i.e. keep `Throwable` but add a comment that states something along the lines of: catching `Throwable` is intentional here as we want to continue the execution even if a particular recommender fails.

Reply:
Catching Throwable is intentional here as we want to continue the execution even if a particular recommender fails.